src/Nethermind/Nethermind.Xdc.Test/XdcHeaderValidatorTests.cs

```diff
@@ -65,6 +65,17 @@ public static IEnumerable<object[]> HeaderTestCases()
         blockHeaderBuilder.WithUnclesHash(Hash256.FromBytesWithPadding([0x01]));
         yield return [blockHeaderBuilder, false];
 
+        //Invalid difficulty
+        blockHeaderBuilder = CreateValidHeader();
+        blockHeaderBuilder.WithDifficulty(2);
+        yield return [blockHeaderBuilder, false];
+
+        //Invalid total difficulty
+        blockHeaderBuilder = CreateValidHeader();
+        blockHeaderBuilder.WithDifficulty(2);
+        blockHeaderBuilder.WithTotalDifficulty(1);
```

Comment:
Difficulty is already invalid here, it'll return false no matter the total difficulty set.
Also these two calls to the builder can be made fluently 🙂